### DISCUSSION TITLE
ECONREFUSED 172.20.0.5:5984 http://couchdb-service:5984/global-db/

### DISCUSSION THREAD

**Discussion by syedkhairi** [Budibase/budibase#8458] [Help]
Hello,

Just followed the instructions [here](https://docs.budibase.com/docs/docker-compose) to install self-host on my Linux Debian.

Linux 5.18.0-3-amd64
AMD EPYC 7542 32-Core Processor

Docker version 20.10.21, build baeda1f
Docker Compose version v2.12.2
docker-compose version 1.27.4, build 40524192
docker-py version: 4.3.1
CPython version: 3.7.7
OpenSSL version: OpenSSL 1.1.0l  10 Sep 2019

However encountered the following error:

> bbapps                                    | Budibase running on {"address":"::","family":"IPv6","port":4002}
> bbapps                                    | Checking migrations
> bbapps                                    | FetchError: request to http://couchdb-service:5984/global-db/ failed, reason: connect ECONNREFUSED 172.20.0.3:5984
> bbapps                                    |     at ClientRequest.<anonymous> (/app/node_modules/node-fetch/lib/index.js:1491:11)
> bbapps                                    |     at ClientRequest.emit (events.js:412:35)
> bbapps                                    |     at ClientRequest.emit (domain.js:475:12)
> bbapps                                    |     at Socket.socketErrorListener (_http_client.js:475:9)
> bbapps                                    |     at Socket.emit (events.js:400:28)
> bbapps                                    |     at Socket.emit (domain.js:475:12)
> bbapps                                    |     at emitErrorNT (internal/streams/destroy.js:106:8)
> bbapps                                    |     at emitErrorCloseNT (internal/streams/destroy.js:74:3)
> bbapps                                    |     at processTicksAndRejections (internal/process/task_queues.js:82:21) {
> bbapps                                    |   type: 'system',
> bbapps                                    |   errno: 'ECONNREFUSED',
> bbapps                                    |   code: 'ECONNREFUSED'
> bbapps                                    | }
> bbapps                                    | [Tenant: default] [Migration: user_email_view_casing] [DB: global-db] Error:  FetchError: request to http://couchdb-service:5984/global-db/ failed, reason: connect ECONNREFUSED 172.20.0.3:5984
> bbapps                                    |     at ClientRequest.<anonymous> (/app/node_modules/node-fetch/lib/index.js:1491:11)
> bbapps                                    |     at ClientRequest.emit (events.js:412:35)
> bbapps                                    |     at ClientRequest.emit (domain.js:475:12)
> bbapps                                    |     at Socket.socketErrorListener (_http_client.js:475:9)
> bbapps                                    |     at Socket.emit (events.js:400:28)
> bbapps                                    |     at Socket.emit (domain.js:475:12)
> bbapps                                    |     at emitErrorNT (internal/streams/destroy.js:106:8)
> bbapps                                    |     at emitErrorCloseNT (internal/streams/destroy.js:74:3)
> bbapps                                    |     at processTicksAndRejections (internal/process/task_queues.js:82:21) {
> bbapps                                    |   type: 'system',
> bbapps                                    |   errno: 'ECONNREFUSED',
> bbapps                                    |   code: 'ECONNREFUSED'
> bbapps                                    | }
> 

No changes were made for the `docker-compose.yaml` or `.env` file.

I've tried [this](https://github.com/Budibase/budibase/discussions/4269#discussioncomment-3794797), in case it is related even though a separate issue - still not working.

Please advise.

**Comment by melohagan** [comment]
Hey @syedkhairi 

Are you still having issues with this? If you look at the **couchdb-service** logs, do you see any errors?

**Comment by kkazakov** [comment]
I'm currently having the same issue. New installation on new system.

The logs from couchbase are:

`No name found in ERL_FLAGS  -setcookie monster`
`No -name found in vm.args. Using couchdb@127.0.0.1`

I had only changed the secrets.

**Reply by melohagan** [reply]
Hey @kkazakov 

Could you tell me a bit more about your system please? What chipset are you running? Could you try to recall the steps that led to this as precisely as you can please?

**Reply by kkazakov** [reply]
It's a personal Proxmox VE server, running on i5 6th generation with 32 gb of ram.
I was starting the docker-compose in privileged LXC alpine linux container.

Usually that's how I run all custom docker containers / docker-compose setups. I have about 20-25 different services running at the moment, based on the same alpine image.

Perhaps Couchbase needs some custom set-up regarding LXCs, I don't know ...

**Reply by mike12345567** [reply]
Hi @kkazakov - just to check is this a fresh install of Budibase? If thats the case, can you try removing the docker volumes/containers created by compose and re-creating?

**Reply by kkazakov** [reply]
Yes, that's a fresh installation. And I already cleaned up everything ( **docker system prune -a -f**  after stopping everything ) and started from scratch, but the result was exactly the same.

**Reply by melohagan** [reply]
`docker volume prune` works for me, although it will of course wipe any of your existing data.

**Reply by mike12345567** [reply]
@kkazakov - thats very odd - I've never seen anything like that before - my best guess is that its related to the CouchDB container running inside an LXC - just to check as well, you're using relatively up to date versions of docker/docker-compose as well?

It might be worth trying our single image instead of the docker-compose installation, generally in any more niche installations this works better. https://docs.budibase.com/docs/docker

**Reply by kkazakov** [reply]
Maybe that's related to the LXC / docker combination, I've had strange problems before. I'll try the docker image and let you know. Otherwise - it's latest alpine with latest updates.

**Reply by kkazakov** [reply]
I had no problems running it via the docker command.
